packages/odata-service-inquirer/src/prompts/datasources/sap-system/system-selection/prompt-helpers.ts

```diff
@@ -85,11 +88,7 @@ export async function connectWithBackendSystem(
             ));
             // If authentication failed with existing credentials the user will be prompted to enter new credentials.
             // We log the error in case there is another issue (unresolveable) with the stored backend configuration.
-            if (
-                errorType === ERROR_TYPE.AUTH &&
-                typeof backendSystem.username === 'string' &&
-                typeof backendSystem.password === 'string'
-            ) {
+            if (errorType === ERROR_TYPE.AUTH) {
```

Comment:
We shouldnt be validating if there are no credentials to validate hence the conditions that have been removed? Note that this code is also used by new system flows probably and so these conditions are relevant in those cases